src/core/src/package_managers/ZypperPackageManager.py

```diff
@@ -79,6 +87,25 @@ def invoke_package_manager(self, command):
             self.set_package_manager_setting(Constants.PACKAGE_MGR_SETTING_REPEAT_PATCH_OPERATION, True)
         return out
 
+    def get_process_tree_from_package_manager_msg(self, message):
+        # Find pid xxxxx within output string
+        regex = re.compile(' pid \d+ ')
+        pid_substr_search = regex.search(message)
+        if pid_substr_search is None:
+            return None
+
+        # Extract just pid text from substring
+        regex = re.compile('\d+')
+        pid_search = regex.search(pid_substr_search.group())
```

Comment:
Since group returns a match, shouldn't it be used after we run either a match or search on the string? re.compile just prepares the pattern we need to search/match. regex.search() is actually going to find that pattern in the string. 

regex.search(pid_substr_search.group())
group used before regex.search is not achieving anything, since it doesn't try to match any pattern with the string during group and before search

Reply:
Yep, it should be used after we run a match or search on the string, and it is in this case. `pid_substr_search` is the search result from using `regex.search` on the original string that was passed in, so group is needed to extract the string found.

Above this code segment are these these lines:
```py
regex = re.compile(' pid \d+')
pid_substr_search = regex.search(message)
```

Reply:
Resolved offline